### ISSUE TITLE
user_warning "Dropping gradients at call to FUNC" is not actionable

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#4055]
When building with the ml-autoscheduler, you'll often get a lot of buildspam of the form:

```
Warning:
Dropping gradients at call to shift_right
Warning:
Dropping gradients at call to reinterpret
Warning:
Dropping gradients at call to bitwise_or
Warning:
Dropping gradients at call to shift_left
```

(etc)

Although emitted as a "user" warning, it's likely that the typical user of the autoscheduler has no idea what this is actually warning about, whether they should be concerned, or how they can fix the warning. If we can't upgrade the text to something user actionable, we should remove it (or downgrade it to debug(1) or lower).

**Comment by abadams** [comment]
@BachiLi  Do you think we need an explicit drop_gradients intrinsic? Or should we just silence this? What do you think it most usable for people using autodiff?

**Comment by abadams** [comment]
(the intrinsic would drop gradients silently)

**Comment by BachiLi** [comment]
It might not be a good idea to silence it just yet. In this case, I believe the dropping warning comes from the fast_log() calls, which is not very obvious that gradients would be dropped. Let the warning be there by default and have an option to turn it off might be better.

**Comment by steven-johnson** [comment]
> It is not very obvious that some gradients would be dropped by just calling an approximate log function.

Indeed, quite so! Which is the entire point of my complaint :-) The warning needs to be made actionable somehow.

**Comment by abadams** [comment]
fast_log should probably be lowered later than it is, like lerp
